Domains/Smartlighting/LightSchedules.md

```diff
@@ -0,0 +1,286 @@
+The `Time` value needs to be formatted in a way the [protocol implementations](#protocol-implementations-for-light-schedules)
+can handle. For the currently listed implementations, you should be fine when you use a format from:
+- `hh:mm`;
+- `hh:mm:ss`;
+- `hh:mm:ss.SSS`
+
+With `hh` from `00` to `23`, `mm` from `00` to `59`, `ss` from `00` to `59` and `SSS` from `000` to `999`.<br>
+Some protocols may accept more precise time formats than they support. The IEC61850 implementation for
+instance, will silently apply only the hours and minutes from any of the formats listed above.
+
+### Astronomical Time
+
+For `ActionTime` `SUNRISE` or `SUNSET` with `TriggerType` `ASTRONOMICAL` the calculated astronomical
+sunrise or sunset time (by the switching device, based on its longitude and latitude) will be used to
+determine the switching moment.
+
+#### Astronomical Offsets
+
+If an astronomical offset is configured, it has to be added to the calculated astronomical time to
+determine the time to be used as the switching moment.<br>
+For positive offset values, the astronomical time for the switching moment will be the configured
+amount of minutes after the calculated astronomical sunrise or sunset time, while for negative values
+the astronomical time used will be the number of minutes before the calculated astronomical sunrise
+or sunset time.
+
+**Astronomical Sunrise Offset**
+
+The astronomical sunrise offset is applied with entries with `ActionTime` `SUNRISE` and
+`TriggerType` `ASTRONOMICAL`.<br>
+The following picture is an example of switching off at 07:30; the calculated astronomical sunrise
+(say at 07:15 for the day shown) plus 15 minutes (configured as `AstronomicalSunriseOffset` `15`).
+
+![Switching off at astronomical sunrise with 15 minutes offset](./astronomical-time-sunrise-with-offset.png)
+
+**Astronomical Sunset Offset**
+
+The astronomical sunset offset is similar to the [astronomical sunrise offset](#astronomical-sunrise-offset),
+except that it is applied with entries with `ActionTime` `SUNSET` and `TriggerType` `ASTRONOMICAL`.
+
+### Astronomical Time With Sensor
+
+For `ActionTime` `SUNRISE` or `SUNSET` with `TriggerType` `LIGHT_TRIGGER` the calculated astronomical
+sunrise or sunset time will be used as a reference time with a trigger window to determine the
+switching moment.<br>
+The astronomical time itself is calculated in the same way as with [astronomical time](#astronomical-time)
+(without light sensor input).<br>
+Switching happens within a configured [trigger window](#trigger-window) around the astronomical time,
+at a moment that is influenced by a signal from a light sensor.
+
+#### Trigger Window
+
+The `TriggerWindow` with its `minutesBefore` and `minutesAfter` defines a window of time around an
+astronomical sunrise or sunset time with sensor.<br>
+Switching will occur at the start of the window when light sensor input is received before the window.<br>
```

Comment:
Originally the devices were supposed to only listen for setTransition messages within the trigger window and switch at the end of the trigger window as fallback. Currently however the Elster and Kaifa devices behave differently. Probably due to a change in requirements, the Elster device will now listen for a setTransition message even before the trigger window and switch at the beginning of the window when such a  message is received, while the Kaifa is still switching at the end of the trigger window when the setTransition message is not received within the trigger window.

Reply:
I based this description on the documentation I have seen for how switching is supposed to work on the Kaifa devices.
The documentation should in my opinion reflect how the switching is supposed to work.
I am not sure where it would be best (if at all desirable) to document that behavior of switching devices may deviate.

Reply:
Bart's text is correct. It conforms to the original requirements stating "In the morning, the night lights are switched off based on a message of the OSGP coming from a light measurement. This message must fit within the window defined around the astronomic sunrise. In the figure, that is moment B. If the light measurement arrives earlier than indicated by the window (moment A), the public lighting must be switched off at the start of the window. If the light measurement does not arrive before or during the window (moment C), the public lighting is switched off at the end of the window.